src/interfaces/Client/IClientInvoker.ts

```diff
@@ -12,7 +12,8 @@ limitations under the License.
 */
 
 import { HttpMethod } from "../../enum/HttpMethod.enum";
+import { InvokerParamsType } from "../../types/InvokerParams.type";
 
 export default interface IClientInvoker {
-    invoke(appId: string, methodName: string, method: HttpMethod, data?: object): Promise<object>;
+    invoke(appId: string, methodName: string, method: HttpMethod,  params?:InvokerParamsType): Promise<object>;
```

Comment:
This is a breaking change, we cannot release it in the next minor version update. A better option will be to add a new optional parameter `options: InvokerOptions` that contains `headers`. That way it won't break anything and will be extensible if we want to add new things to options. 

What do you think @XavierGeerinck?

Reply:
@shubham1172  can I go ahead with the above approach  or should I wait for @XavierGeerinck  comment

Reply:
Let's give them some time to respond.

Reply:
Agreed that it is a breaking change. I do agree as well that we need to have it everywhere. So for me we can do this in 3.0.0 but there is no ETA yet on 3.0.0. So in the meantime I suggest to add it as a 4th param if it's urgent and create an issue for 3.0.0 where we deprecate the older ways

Reply:
@shubham1172 I have pushed the changes .you can review them.

Reply:
@salmankhan-prs please fix DCO https://github.com/dapr/js-sdk/pull/373/checks?check_run_id=8508406778